buffer/src/main/java/io/netty/buffer/api/ByteCursor.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2021 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.netty.buffer.api;
+
+import io.netty.util.ByteProcessor;
+
+/**
+ * The ByteCursor scans through a sequence of bytes.
+ * This is similar to {@link ByteProcessor}, but for external iteration rather than internal iteration.
+ * The external iteration allows the callers to control the pace of the iteration.
+ * The API includes methods for reading {@code long}s as a batch of 8 bytes.
+ * The long values are always in big-endian format, so that the highest-order byte in the long value, contain the byte
+ * that would otherwise have been returned by the next call to {@link #getByte()}.
+ */
+public interface ByteCursor {
```

Comment:
- Why not other primitive accessors, e.g. short, medium, unsigned?
- Little endian support?

Reply:
The conceptual model is to process a stream of bytes. Reading longs is a batch optimisation.

Reply:
I foresee users want other accessor methods I mentioned, at least as a convenient shortcut for multiple `getByte()` calls.

Reply:
If you think about short/int use cases probably makes sense, while `byte[Long.BYTES]` probably won't worth.
I remember this one has been introduced to ease porting batch processing a la https://github.com/netty/netty/pull/10737 
(recently I just talk about this PR, sorry about it :P )

Reply:
The "interval" idea would still have to bounds check to make sure the cursor is within the interval.